alicloud/config.go

```diff
@@ -399,6 +402,57 @@ func (c *Config) getAuthCredential(stsSupported bool) auth.Credential {
 	return credentials.NewAccessKeyCredential(c.AccessKey, c.SecretKey)
 }
 
+func (client *AliyunClient) AccountId() (string, error) {
+	client.accountIdMutex.Lock()
+	defer client.accountIdMutex.Unlock()
+
+	if client.accountId == "" {
+		log.Printf("[DEBUG] account_id not provided, attempting to retrieve it automatically...")
+		identity, err := client.GetCallerIdentity()
+		if err != nil {
+			return "", err
+		}
+		if identity.AccountId == "" {
+			return "", GetNotFoundErrorFromString("Caller identity doesn't contain any AccountId.")
+		}
+		log.Printf("[DEBUG] account_id retrieved with success.")
+		client.accountId = identity.AccountId
+	}
+	return client.accountId, nil
+}
+
+func (client *AliyunClient) Fcconn() (*fc.Client, error) {
+	client.fcconnMutex.Lock()
+	defer client.fcconnMutex.Unlock()
```

Comment:
fcconnMutex seems as accountIdMutex.
In addition, I can not understand why you defined a new Fcconn and not modify it.

Reply:
I use fcconnMutex for the same reason as accountIdMutex: for thread safety. In addition I use a different mutex for fcconn because it is not systematically used with account_id: for example the resource alicloud_router_interface_connection uses the account_id but not fcconn.

Reply:
One more thing: the field `fcconn` is set in https://github.com/marcplouhinec/terraform-provider/blob/0545a8251e84d89049f339a03d034652c73c541e/alicloud/config.go#L442